draft-ietf-ppm-dap.md

```diff
@@ -1157,6 +1157,7 @@ struct {
 struct {
   ReportMetadata report_metadata;
   opaque public_share<0..2^32-1>;
+  Extension public_extensions<0..2^32-1>;
```

Comment:
This change would be smaller if you added the extensions to ReportMetadata.  Any reason why you chose not to?

Reply:
I should have added.  4 gigabytes is an awful lot.  Do I have to start sharpening my varint axe?  There's a lot of overhead in this protocol that would be trivially cut with a single slice.

Reply:
+1, let's move this to the report metadata. This will also help with code movement in daphne.

Reply:
> I should have added. 4 gigabytes is an awful lot. Do I have to start sharpening my varint axe? There's a lot of overhead in this protocol that would be trivially cut with a single slice.

Using varints for length specifiers is not (AFAICT) supported by TLS syntax we currently use; we'd probably extend our syntax somehow to specify this. I'd need to do a little digging to see what varint formats are currently specified/standardized; a very quick glance does reveal anything. (Of course, if we can't lean on prior work we could certainly specify something ourselves.)

Reply:
Oh, I should also add: I'm in favor of a 2-byte prefix, for consistency with the private extensions.